src/ceph-volume/ceph_volume/util/prepare.py

```diff
@@ -64,7 +64,7 @@ def create_id(fsid, json_secrets, osd_id=None):
         '-i', '-',
         'osd', 'new', fsid
     ]
-    if check_id(osd_id):
+    if not check_id_exists(osd_id) and osd_id is not None:
```

Comment:
It would be better to swap these around, so that there is no need to contact the monitors every time if the `osd_id` is a `None`:

    if osd_id is not None and not osd_id_exists(osd_id):